### PR TITLE
Updated states.md for Bevy v0.10

### PR DESCRIPTION
**Pull request by Connor-McMillin**
First PR for anything Bevy related so I'm sure there are some things to change in here. I left some TODOs in areas that I wasn't sure if there were changes between 0.9 and 0.10 and hope we can answer those TODOs in the PR conversation

**Changes**
- Created basics.rs for Bevy v0.10 based on basics.rs for Bevy v0.9 and updated all examples so they compile with Bevy v0.10.1
    - `esc_to_menu` differs from the relevant example implementation given in `start_game` in the [Bevy v0.10 announcement](https://bevyengine.org/news/bevy-0-10/#run-conditions). I couldn't find an `Interaction::Pressed` in documentation and rust-analyzer did not recognize it either. I found [an example](https://github.com/bevyengine/bevy/blob/44a365d5403dbdb41862f169c6451457fba1ec98/examples/input/mouse_grab.rs#L26-L29) using `Res<Input<KeyCode>>` from the repo which is where I got my implementation for checking escape key press from.
- Updated documentation in states.md to reflect changes from v0.9 to 0.10
    -  Removed state stack section because states are [no longer stack based](https://bevyengine.org/news/bevy-0-10/#simpler-states)
    - Example references are now to v0.10 code
    - Added TODOs in sections where I need other's input

### CODE REVIEW

File: src/code010/src/basics.rs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    app_state.set(AppState::InGame);
+}
+// ANCHOR_END: change-state
+
+// ANCHOR: state-input-clear
+fn esc_to_menu(
+    key: Res<Input<KeyCode>>,
+    mut next_state: ResMut<NextState<AppState>>,
+) {
+    if key.just_pressed(KeyCode::Escape) {
```
Comment (Connor-McMillin) [inline src/code010/src/basics.rs:40]: Here is the esc_to_menu function referenced in the PR notes

File: src/programming/states.md [hunk 2]
```diff
@@ -34,23 +34,25 @@ system ordering][cb::system-order].
 This can also be useful with [Plugins][cb::plugin]. Each plugin can add
 its own set of systems to the same state.
 
+// TODO: Is this paragraph still true?
```
Comment (Connor-McMillin) [inline src/programming/states.md:37]: Are states still implemented using cb::runcriteria under the hood?

**Comment by Connor-McMillin** [comment]
@inodentry I could use some input on this draft PR. I'm fairly new to Bevy, but figured I could make this update to help out other new folks looking for v0.10 state info

**Comment by inodentry** [comment]
Hey. Thank you for expressing interest in updating Bevy Cheatbook. Your heart is in the right place, but I can't accept PRs like this, I hope you understand. I don't want you to waste your time and energy.

Bevy Cheatbook is unofficial (not affiliated with the Bevy project). It is my personal project. While I sometimes accept community contributions (especially smaller things like fixes and tweaks), I don't have the capacity to review larger PRs like this one. Having to mentor you through this draft, go back and forth reviewing and teaching you stuff, is going to take more of my time than just doing it myself.

Also, I'm currently working on updating the book to Bevy 0.10 myself, and I already have WIP stuff for the programming chapter (incl the new states, schedules, run conditions etc.) on my computer, that I haven't published yet. Your work is redundant with that.

Also, given how much new stuff there is with 0.10, I feel the need to teach everything in a bit of a different way, and restructure the book somewhat. Part of that is already public, and more will be coming soon.

Internal project organization is also changing. `basics.rs` is a holdover from back in 2021 when the book started and I put all the example code in one Rust file. I don't want to carry it over to 0.10. As I am reworking the book, I am using the opportunity to reorganize and improve the code structure. Your work here is also redundant with that.

I will close this PR. Again, your interest is much appreciated. ❤️  I hope you don't feel offended over my rejection. It's just that this PR is not coordinated with the current project direction.

**Comment by Connor-McMillin** [comment]
No worries at all. Thanks for the Cheatbook and I'm excited to see the 0.10 updates! ❤️

**Pull request closed without merging**